spec.html

```diff
@@ -30263,18 +30266,21 @@ <h1>
               1. Let _parameterSym_ be the grammar symbol |FormalParameters[+Yield, +Await]|.
               1. Let _fallbackProto_ be *"%AsyncGeneratorFunction.prototype%"*.
             1. Let _argCount_ be the number of elements in _parameterArgs_.
+            1. Let _parametersStrings_ be a new empty List.
+            1. For each element _arg_ of _parameterArgs_, do
+              1. Append ? ToString(_arg_) to _parametersStrings_.
             1. Let _P_ be the empty String.
             1. If _argCount_ > 0, then
-              1. Let _firstArg_ be _parameterArgs_[0].
-              1. Set _P_ to ? ToString(_firstArg_).
+              1. Set _P_ to _parametersStrings_[0].
               1. Let _k_ be 1.
               1. Repeat, while _k_ &lt; _argCount_,
-                1. Let _nextArg_ be _parameterArgs_[_k_].
-                1. Let _nextArgString_ be ? ToString(_nextArg_).
+                1. Let _nextArgString_ be _parametersStrings_[_k_].
                 1. Set _P_ to the string-concatenation of _P_, *","* (a comma), and _nextArgString_.
                 1. Set _k_ to _k_ + 1.
-            1. Let _bodyString_ be the string-concatenation of 0x000A (LINE FEED), ? ToString(_bodyArg_), and 0x000A (LINE FEED).
-            1. Let _sourceString_ be the string-concatenation of _prefix_, *" anonymous("*, _P_, 0x000A (LINE FEED), *") {"*, _bodyString_, and *"}"*.
+            1. Let _bodyString_ be ? ToString(_bodyArg_).
```

Comment:
Note that this changes the String used to define _body_. I don't *think* it matters, but I'm not sure with this AO.

Reply:
_sourceString_ is still the same anyway, right?

Reply:
It seems a bit odd to me, that the concatenated surrounding `\n`s were passed to ParseText in the first place, but I don't think it makes a difference. I haven't checked, but I'd guess that engines did it this way already, otherwise line numbers in resulting SyntaxErrors would be off by one.

Reply:
Re the inserted newlines:

They first appeared, in _sourceText_, when that was introduced in PR #697.

Then PR #1479 modified things so that they *also* appeared in _bodyText_. And that's the only substantive change that that PR made, so this change would undo that PR.

I don't entirely understand the issue, but it seems to have something to do with HTML comments and web reality.

Reply:
Oh yes, https://tc39.es/ecma262/#prod-annexB-SingleLineHTMLCloseComment requires a leading line terminator.

Then the order to preserve the correct behavior should be:
- call ToString(...)
- pass the result to the host hook
- add the `\n`s before and after
- pass it to ParseText(StringToCodePoints(_bodyString_), _bodySym_)

Reply:
Yeah, if the host hook doesn't want the newline-encrusted string.

Reply:
I've pushed a new update that should address this discrepancy. We now make a distinction between _bodyString_ (without added newlines) and _bodyParseString_ (with added newlines).